packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -598,6 +598,24 @@ function updateSimpleMemoComponent(
       (__DEV__ ? workInProgress.type === current.type : true)
     ) {
       didReceiveUpdate = false;
+
+      // The props are shallowly equal. Reuse the previous props object, like we
+      // would during a normal fiber bailout.
+      //
+      // We don't have strong guarantees that the props object is referentially
+      // equal during updates where we can't bail out anyway — like if the props
+      // are shallowly equal, but there's a local state or context update in the
+      // same batch.
+      //
+      // However, as a principle, we should aim to make the behavior consistent
+      // across different ways of memoizing a component. For example, React.memo
+      // has a different internal Fiber layout if you pass a normal function
+      // component (SimpleMemoComponent) versus if you pass a different type
+      // like forwardRef (MemoComponent). But this is an implementation detail.
+      // Wrapping a component in forwardRef (or React.lazy, etc) shouldn't
+      // affect whether the props object is reused during a bailout.
+      workInProgress.pendingProps = nextProps = prevProps;
```

Comment:
nice! so in a way, my fix was same(ish):
https://github.com/facebook/react/blob/bd13426453866c50b27a813d26e9396aa34dbf8b/packages/react-reconciler/src/ReactFiberBeginWork.new.js#L597-L599

this one is "just" applied at a more proper, generic, level 👍

Reply:
yeah it's essentially the same, I just kept the `shallowEqual` call where it already was